### PR TITLE
Mandos esdt

### PR DESCRIPTION
**Pull request by dorin-iancu**
ESDT support added to mandos tests + crowdfunding contract esdt example added

### CODE REVIEW

File: mandos/src/account.rs [hunk 1]
```diff
@@ -61,12 +72,43 @@ impl CheckStorage {
 	}
 }
 
+#[derive(Debug)]
+pub enum CheckEsdt {
```
Comment (andrei-marinica) [inline mandos/src/account.rs:76]: You can avoid having to use an extra Option, by having DefaultStar, same as with CheckValue. DefaultStar is identical to Star in behaviour, but it occurs when nothing is written in the json. They are separate just so we can reconstruct the mandos the other way around, no other function. DefaultStar would be the Option::None.

File: elrond-wasm-debug/src/execute_mandos.rs [hunk 2]
```diff
@@ -373,6 +437,105 @@ fn check_state(accounts: &mandos::CheckAccounts, state: &mut BlockchainMock) {
 					);
 				}
 			}
+
+			match &expected_account.esdt {
+				Some(CheckEsdt::Equal(eq)) => {
```
Comment (andrei-marinica) [inline elrond-wasm-debug/src/execute_mandos.rs:442]: There is a Checkable trait, that is used precisely for this. Implementing Checkable for CheckEsdt gives you the "check" method, where you can move the checking logic, with the match and all.
The point is to have less code here, where there is already plenty, and have at least the checking done somewhere else.

File: mandos/src/account.rs [hunk 3]
```diff
@@ -61,12 +72,43 @@ impl CheckStorage {
...
+			),
+		}
+	}
+}
+
+impl CheckEsdt {
+	pub fn is_star(&self) -> bool {
+		matches!(self, CheckEsdt::Star)
+	}
+}
```
Comment (andrei-marinica) [inline mandos/src/account.rs:103]: You should also implement the Checkable trait here, it takes away code from other, more crowded places.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: elrond-wasm-debug/src/blockchain_mock.rs [hunk 4]
```diff
@@ -181,6 +199,68 @@ impl BlockchainMock {
...
+		let esdt_balance = esdt.get_mut(esdt_token_name).unwrap_or_else(|| {
+			panic!(
+				"Account {} has no esdt tokens with name {}",
+				address_hex(&address),
+				String::from_utf8(esdt_token_name.to_vec()).unwrap()
+			)
+		});
+
+		assert!(
+			*esdt_balance >= *value,
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply elrond-wasm-debug/src/blockchain_mock.rs:227]: Nope, that gives the following error: 
mismatched types
expected mutable reference `&mut num_bigint::BigUint`
               found reference `&num_bigint::BigUint`

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**